usr/share/rear/prep/Linux-i386/370_detect_secure_boot.sh

```diff
@@ -0,0 +1,34 @@
+test "$SECURE_BOOT_BOOTLOADER" && return 0
+
+# Note:
+# this is more of a hack than a really good solution.
+# The good solution would check the EFI variables for the bootloader
+# that is used to boot the system. But this is not implemented yet for
+# secure boot.
+#
+# The code in usr/share/rear/rescue/default/850_save_sysfs_uefi_vars.sh
+# could be used as a starting point for this, however currently
+# it tries to read the actual boot loader# used only as a last resort
+# if well-known files are not found.
+#
+if type -p mokutil ; then
+    PROGS+=( mokutil )
+    local secureboot_status
+    if secureboot_status=$(mokutil --sb-state 2>&1) ; then
+        if grep -q "SecureBoot enabled" <<<"$secureboot_status" ; then
+            # Check first for arch-specific and then generic shim file, nullglob will give us the first one found and we ignore the others.
+            # shellcheck disable=SC2206
+            SECURE_BOOT_BOOTLOADER=( /boot/efi/EFI/*/shim$EFI_ARCH.efi /boot/efi/EFI/*/shim.efi )
+            # shellcheck disable=SC2128
+            LogPrint "Secure Boot auto-configuration using '$SECURE_BOOT_BOOTLOADER' as UEFI bootloader"
+        else
```

Comment:
@gdha @pcahyna @jsmeix I could add here a check for empty `SECURE_BOOT_BOOTLOADER` and error out with an explanation that secure boot is active but we couldn't find a shim

Reply:
@schlomo 
at least for now I think we should keep it simple
so I think you may do it as you currently like most.

Personally I prefer during "rear mkrescue"
to better error out when things look wrong cf.
"Try hard to care about possible errors" in
https://github.com/rear/rear/wiki/Coding-Style
even if this means a false error out in special cases
but then users will tell us about their special cases
so we can improve things together with our users.

In this case the simple workaround for special cases
is that the user specifies SECURE_BOOT_BOOTLOADER
so a false error out here does not cause real harm.

An addedum only FYI:
On the other hand I prefer to not error out during
"rear recover" after the backup was restored, see
https://github.com/rear/rear/issues/2612#issuecomment-842304271

Reply:
@schlomo yes please add it, I think it is not too much work and it will help catching unexpected situations.